src/Data/List/Permutation.agda

```diff
@@ -0,0 +1,134 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Permutation
+------------------------------------------------------------------------
+
+module Data.List.Permutation where
+
+open import Relation.Binary.PropositionalEquality as ≡
+  using (_≡_ ; refl ; setoid ; _≢_ ; isEquivalence)
+  renaming (trans to ≡-trans)
+open import Relation.Binary
+
+open import Data.List
+open import Data.List.Any
+
+open import Function
+
+open import Data.Empty
+open import Data.Product using (Σ ; _,_ ; proj₁ ; proj₂ ; _×_ ; ∃ ; ∃₂)
+
+open import Data.List.Properties
+
+module _ {a} {A : Set a} where
+
+  -- an inductive definition of permutation
+
+  data perm : Rel (List A) a where
+    unit  : ∀ {l} → perm l l
```

Comment:
I would call this constructor `refl` for consistency with the rest of the standard library.